src/main/java/com/mongodb/spark/sql/connector/config/WriteConfig.java

```diff
@@ -67,6 +71,41 @@ public String toString() {
     }
   }
 
+  /** The convert json configuration. */
+  public enum ConvertJson {
+    /** No conversion of string types */
+    FALSE("false"),
+    /** Try to parse any string as a json value */
+    ANY("any"),
+    /** Try to parse any string as a json value */
+    OBJECT_OR_ARRAY_ONLY("objectOrArrayOnly");
+
+    private final String value;
+    private static final String TRUE = "true";
+
+    ConvertJson(final String operationType) {
+      this.value = operationType;
+    }
+
+    static ConvertJson fromString(final String jsonType) {
+      for (ConvertJson convertJsonType : ConvertJson.values()) {
+        if (jsonType.equalsIgnoreCase(TRUE)) {
+          LOGGER.warn("{}: 'true' is deprecated. Use: '{}' instead.", CONVERT_JSON_CONFIG, ANY);
+          return ANY;
+        }
+        if (jsonType.equalsIgnoreCase(convertJsonType.value)) {
+          return convertJsonType;
+        }
+      }
+      throw new ConfigException(format("'%s' is not a valid Convert Json Type", jsonType));
+    }
```

Comment:
- There is no need to check `if (jsonType.equalsIgnoreCase(TRUE))` in a loop.
- Instead of looping, we can use the existing `ConvertJson.valueOf` method.


```suggestion
    static ConvertJson fromString(final String jsonType) {
      if (jsonType.equalsIgnoreCase(TRUE)) {
        LOGGER.warn("{}: '{}' is deprecated. Use: '{}' instead.", CONVERT_JSON_CONFIG, TRUE, ANY);
        return ANY;
      }
      try {
        return ConvertJson.valueOf(jsonType.toUpperCase(Locale.ROOT));
      } catch (IllegalArgumentException e) {
        throw new ConfigException(format("'%s' is not a valid Convert Json Type", jsonType), e);
      }
    }
```